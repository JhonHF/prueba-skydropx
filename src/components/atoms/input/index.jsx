import React from "react";
import { useField } from "formik";

export const Input = ({ label, name, className, ...rest }) => {
  const [input, meta] = useField(name);

  const errorClass = meta.error ? "is-invalid" : "";

  return (
    <div className={`form-group ${className}`}>
      {label && (
        <label htmlFor={name} className="form-label">
          {label}
        </label>
      )}
      <input
        className={`form-control ${errorClass}`}
        id={name}
        {...input}
        {...rest}
      />
      {meta.error && <div className="invalid-feedback">{meta.error}</div>}
    </div>
  );
};
