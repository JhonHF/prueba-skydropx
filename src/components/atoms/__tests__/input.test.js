import React from "react";
import { shallow } from "enzyme";
import { FormikProvider } from "formik";

import { Input } from "../input";

describe("Input component", () => {
  describe("GIVEN a set of valid props", () => {
    const testProps = {
      name: "name",
      className: "form-control",
    };

    describe("WHEN the component is rendered without props", () => {
      const shallowWrapper = shallow(
        <FormikProvider>
          <Input {...testProps} />
        </FormikProvider>
      );

      it("THEN should display a regular AddPrestation", () => {
        expect(shallowWrapper).toMatchSnapshot();
      });
    });

    describe("WHEN the card is selected", () => {
      const customProps = {
        ...testProps,
        label: "Label",
      };

      const shallowWrapper = shallow(
        <FormikProvider>
          <Input {...customProps} />
        </FormikProvider>
      );

      it("THEN the onClick function is called", () => {
        expect(shallowWrapper.find("Input").props().label).toEqual("Label");
      });
    });
  });
});
