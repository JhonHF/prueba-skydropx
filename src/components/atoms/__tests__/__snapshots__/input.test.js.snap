// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component GIVEN a set of valid props WHEN the component is rendered without props THEN should display a regular AddPrestation 1`] = `
<Input
  className="form-control"
  name="name"
/>
`;
